pkg/config/viper.go

```diff
@@ -28,7 +28,7 @@ type safeConfig struct {
 	sync.RWMutex
 	envPrefix      string
 	envKeyReplacer *strings.Replacer
-	configEnvVars  []string
+	configEnvVars  map[string]struct{}
```

Comment:
I think a comment here would be valuable to explain why we don't just use an array for future maintenance.